### PR TITLE
Refactor HTTP header parsing and normalize quality handling

### PR DESCRIPTION
**Pull request by dave-shawley**
## Summary
- extract shared cursor-based parsing helpers for HTTP-style headers and move Link parsing to the new parser path
- normalize quality handling across Accept-style parsers and ContentType quality metadata
- tighten header-specific empty-item and malformed-content handling while preserving parse_list as the generic list splitter
- add targeted parser and header regressions, including quoted-list-item unescaping

## Validation
- uv run --no-sync pytest -q
- uv run --no-sync ruff check src/ietfparse tests
- uv run --no-sync pyrefly check src/ietfparse tests
- uv run --no-sync ty check src/ietfparse tests --output-format=concise --exit-zero

## Summary by CodeRabbit

* **New Features**
  * Improved RFC-style Link/header parsing and normalized content-type quality semantics (including string comparisons).

* **Bug Fixes**
  * More robust handling of trailing commas, nested comments, escaped/quoted delimiters, valueless parameters; malformed inputs now raise clearer parsing errors (added MalformedListSegment).

* **Documentation**
  * Expanded changelog and API error docs describing parsing and quality-normalization behavior.

* **Tests**
  * Large suite of new and expanded tests covering parsing edge cases and negotiation ordering.

* **Chores**
  * Added CI recipe and enabled permissive ignores in tooling config.

[Review Change Stack](https://app.coderabbit.ai/change-stack/dave-shawley/ietfparse/pull/52?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: tests/test_datastructure.py [hunk 1]
```diff
@@ -46,6 +47,77 @@ def test_that_content_suffix_is_appended(self) -> None:
...
+    ) -> None:
+        content_type = datastructures.ContentType(
+            'a', 'b', parameters={'q': decimal.Decimal('0.1245')}
+        )
+        self.assertEqual(0.125, content_type.quality)
+
+    def test_that_large_decimal_quality_is_treated_as_highest(self) -> None:
+        content_type = datastructures.ContentType('a', 'b')
+        content_type.quality = decimal.Decimal('1E+100000')
+        self.assertEqual(1.0, content_type.quality)
```
[the remark this answers is not in the document]
Comment (dave-shawley) [reply tests/test_datastructure.py:90]: Addressed in 948dbf6
